c++/src/kj/compat/http.c++

```diff
@@ -2120,37 +2168,56 @@ public:
 
 class HttpFixedLengthEntityWriter final: public kj::AsyncOutputStream {
 public:
-  HttpFixedLengthEntityWriter(HttpOutputStream& inner, uint64_t length)
+  HttpFixedLengthEntityWriter(ResetableRef<HttpOutputStream>& inner, uint64_t length)
       : inner(inner), length(length) {
-    if (length == 0) inner.finishBody();
+    if (length == 0) KJ_ASSERT_NONNULL(inner.tryGet()).finishBody();
   }
   ~HttpFixedLengthEntityWriter() noexcept(false) {
-    if (length > 0 || inner.isWriteInProgress()) {
-      inner.abortBody();
+    auto& i = KJ_UNWRAP_OR(inner.tryGet(), {
+      logInnerClearedWarningOnce();
+      return;
+    });
+    if (length > 0 || i.isWriteInProgress()) {
+      i.abortBody();
     }
   }
 
   Promise<void> write(const void* buffer, size_t size) override {
     if (size == 0) return kj::READY_NOW;
+    auto& i = KJ_UNWRAP_OR(inner.tryGet(), {
+      logInnerClearedWarningOnce();
+      return kj::READY_NOW;
+    });
     KJ_REQUIRE(size <= length, "overwrote Content-Length");
     length -= size;
 
-    return maybeFinishAfter(inner.writeBodyData(buffer, size));
+    return maybeFinishAfter(i.writeBodyData(buffer, size));
+    return kj::READY_NOW;
   }
   Promise<void> write(ArrayPtr<const ArrayPtr<const byte>> pieces) override {
     uint64_t size = 0;
+    auto& i = KJ_UNWRAP_OR(inner.tryGet(), {
+      logInnerClearedWarningOnce();
+      return kj::READY_NOW;
+    });
+
     for (auto& piece: pieces) size += piece.size();
 
     if (size == 0) return kj::READY_NOW;
     KJ_REQUIRE(size <= length, "overwrote Content-Length");
```

Comment:
Nit: I think it'd be preferable to still throw if this precondition doesn't hold. Here and above in the other overload of `write()`.